e2e/interchain_accounts_test.go

```diff
@@ -389,3 +389,165 @@ func (s *InterchainAccountsTestSuite) TestMsgSubmitTx_SuccessfulBankSend_Incenti
+			address, err := s.QueryCounterPartyPayee(ctx, chainB, chainBRelayerWallet.Address, channelOutput.Counterparty.ChannelID)
+			s.Require().NoError(err)
+			s.Require().Equal(chainARelayerWallet.Address, address)
+		})
+
+		t.Run("no incentivized packets", func(t *testing.T) {
+			packets, err := s.QueryIncentivizedPacketsForChannel(ctx, chainA, channelOutput.PortID, channelOutput.ChannelID)
+			s.Require().NoError(err)
+			s.Require().Empty(packets)
+		})
+
+		t.Run("stop relayer", func(t *testing.T) {
+			err := relayer.StopRelayer(ctx, s.GetRelayerExecReporter())
+			s.Require().NoError(err)
+		})
+
+		t.Run("broadcast incentivized MsgSubmitTx", func(t *testing.T) {
+			msgPayPacketFee := &feetypes.MsgPayPacketFee{
+				Fee:             testvalues.DefaultFee(chainADenom),
+				SourcePortId:    channelOutput.PortID,
+				SourceChannelId: channelOutput.ChannelID,
+				Signer:          controllerAccount.Bech32Address(chainA.Config().Bech32Prefix),
+			}
+
+			msgSend := &banktypes.MsgSend{
+				FromAddress: interchainAcc,
+				ToAddress:   chainBAccount.Bech32Address(chainB.Config().Bech32Prefix),
+				Amount:      sdk.NewCoins(testvalues.DefaultTransferAmount(chainB.Config().Denom)),
+			}
+
+			msgSubmitTx, err := intertxtypes.NewMsgSubmitTx(msgSend, ibctesting.FirstConnectionID, controllerAccount.Bech32Address(chainA.Config().Bech32Prefix))
+			s.Require().NoError(err)
+
+			resp, err := s.BroadcastMessages(ctx, chainA, controllerAccount, msgPayPacketFee, msgSubmitTx)
+			s.AssertValidTxResponse(resp)
+			s.Require().NoError(err)
+
+			s.Require().NoError(test.WaitForBlocks(ctx, 1, chainA, chainB))
+		})
+
+		t.Run("there should be incentivized packets", func(t *testing.T) {
+			packets, err := s.QueryIncentivizedPacketsForChannel(ctx, chainA, channelOutput.PortID, channelOutput.ChannelID)
+			s.Require().NoError(err)
+			s.Require().Len(packets, 1)
+			actualFee := packets[0].PacketFees[0].Fee
+
+			s.Require().True(actualFee.RecvFee.IsEqual(testFee.RecvFee))
+			s.Require().True(actualFee.AckFee.IsEqual(testFee.AckFee))
+			s.Require().True(actualFee.TimeoutFee.IsEqual(testFee.TimeoutFee))
+		})
+
+		t.Run("start relayer", func(t *testing.T) {
+			s.StartRelayer(relayer)
+		})
+
+		t.Run("packets are relayed", func(t *testing.T) {
+			packets, err := s.QueryIncentivizedPacketsForChannel(ctx, chainA, channelOutput.PortID, channelOutput.ChannelID)
+			s.Require().NoError(err)
+			s.Require().Empty(packets)
+		})
+
+		t.Run("verify interchain account did not send tokens", func(t *testing.T) {
+			balance, err := chainB.GetBalance(ctx, chainBAccount.Bech32Address(chainB.Config().Bech32Prefix), chainB.Config().Denom)
+			s.Require().NoError(err)
+
+			_, err = chainB.GetBalance(ctx, interchainAcc, chainB.Config().Denom)
+			s.Require().NoError(err)
+
+			expected := testvalues.StartingTokenAmount
+			s.Require().Equal(expected, balance, "tokens should not have been sent as interchain account was not funded")
```

Comment:
Since the interchain account is not funded, shouldn't the balance be zero?

Reply:
this is just because all of the accounts are initialized like this

```golang
chainBAccount := s.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
```

It is not possible to initialize an account with a balance of `0` with the framework as far as I understand.